### PR TITLE
fix broken remote path when using cwd in gulp.src

### PR DESCRIPTION
**Pull request by timaschew**
If you want to upload files within a subfolder like `src` from you local file system but don't want to have this directory on NetSuite in the File Cabinet you can usually pass a `cwd` property to `gulp.src`.

But this breaks the remote path in nscabinet, because the default value for `cwd` is `process.cwd()`. Gulp provides to pass a relative path to the `process.cwd()`, so in this case the path needs to be resolved to an absolute path, because this part will be cut out for the remote path.

### CODE REVIEW

File: src/nscabinet.js [hunk 1]
```diff
@@ -99,6 +97,11 @@ module.exports.download = (files,params) => {
 
         function end() {
             this.emit('end');
+            if (typeof callback === 'function') {
```
Comment (danieljoppi) [inline src/nscabinet.js:100]: create unit test for this block:
- https://coveralls.io/files/1919631033

File: src/nscabinet.js [hunk 2]
```diff
@@ -68,7 +67,7 @@ module.exports.download = (files,params) => {
         files : files ,
         rootpath: params.rootPath
     };
-
+    var result = []
```
Comment (danieljoppi) [inline src/nscabinet.js:70]: missing semicolon

File: src/nscabinet.js [hunk 3]
```diff
@@ -99,6 +97,11 @@ module.exports.download = (files,params) => {
 
         function end() {
             this.emit('end');
+            if (typeof callback === 'function') {
+              callback(result.map(function(file) {
+                return file.path
```
Comment (danieljoppi) [inline src/nscabinet.js:102]: missing semicolon

File: src/nscabinet.js [hunk 4]
```diff
@@ -99,6 +97,11 @@ module.exports.download = (files,params) => {
...
+              }))
```
Comment (danieljoppi) [inline src/nscabinet.js:103]: missing semicolon

File: src/nscabinet.js [hunk 5]
```diff
@@ -79,10 +78,9 @@ module.exports.download = (files,params) => {
                 return;
             }
 
-            data.files = data.files || [];
+            result = data.files = data.files || [];
```
Comment (danieljoppi) [inline src/nscabinet.js:81]:
``` javascript
data.files = data.files || [];
if (data.files.length) {
    result = data.files;
}
```

or 

``` javascript
data.files = data.files || [];
if (data.files.length) {
    result = result.concat(data.files);
}
```

**Comment by danieljoppi** [comment]
execute `gulp test` before push

fix build 
- https://travis-ci.org/suiteplus/nscabinet/jobs/105686437#L2487

``` bash
src/nscabinet.js
   70:20  error  Missing semicolon                                         semi
  101:15  error  Expected indentation of 16 space characters but found 14  indent
  102:17  error  Expected indentation of 18 space characters but found 16  indent
  102:33  error  Missing semicolon                                         semi
  103:18  error  Missing semicolon                                         semi
```

**Comment by timaschew** [comment]
Damn, I didn't want to put the second commit into this pull request, this was just a workaround for #10 

So the second commit should only be a "private" fork, that's why I didn't wrote any semicolons. I started to avoid writing them for ES6 and Nodejs.

Of course for pull requests I adapt the style to the style guide of the author/maintainer.

So I will close this one and create a new PR with only the first commit

**Pull request closed without merging**